jhub_apps/configuration.py

```diff
@@ -124,4 +126,8 @@ def install_jhub_apps(c, spawner_to_subclass, *, oauth_no_confirm=False):
         c.JupyterHub.load_roles = c.JupyterHub.load_roles + services_roles
     else:
         c.JupyterHub.load_roles = services_roles
+    
+    # I considered instantiating startup_apps here b/c I have access to c.JAppsConfig.startup_apps, but JUPYTERHUB_API_URL wasn't set yet which was needed by the HubClient so I moved it to jhub_apps/service/app.py.
```

Comment:
Possible to start the startup apps here.  Any opinions on which place is better @aktech?

Reply:
I don't think it is possible to start apps here since the JupyterHub API would not be up at this point.

Where you have started it is the right place IMO. You just need to make sure JupyterHub API is accessible at the point and there is no deadlock, like potentially: JupyterHub starts FastAPI service, then waiting for the service to be up and the service before its started its waiting the lifespan hook to be ran, which is waiting for JupyterHub API to be up (just an example).

Reply:
I was able to run this PR as is locally without deadlock so I think we're alright atm.

Reply:
Awesome! 🎉